### PR TITLE
[Queues] Migrate to devtools test framework

### PR DESCRIPTION
**Pull request by rakshith91**
Currently test suite uses legacy test framework and unittest. This PR is to migrate the test framework for Queues to the devtools utils which uses pytest.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-queue/tests/asyncqueuetestcase.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        """Synchronous wrapper for async test methods. Used to avoid making changes
+        upstream to AbstractPreparer (which doesn't await the functions it wraps)
+        """
+
+        @functools.wraps(test_fn)
+        def run(test_class_instance, *args, **kwargs):
+            resource_group = kwargs.get("resource_group")
+            location = kwargs.get("location")
+            storage_account = kwargs.get("storage_account")
+            storage_account_key = kwargs.get("storage_account_key")
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/asyncqueuetestcase.py]: You probably don't need to extract all these from kwargs, would it work just be doing:
```
loop.run_until_complete(test_funct(test_class_instance, **kwargs))
```

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py [hunk 2]
```diff
@@ -307,96 +146,6 @@ def assertNamedItemNotInContainer(self, container, item_name, msg=None):
                     repr(item_name), repr(container))
                 self.fail(self._formatMessage(msg, standardMsg))
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/queuetestcase.py]: Now that we're not using `unittest.Testcase`, I don't think `self.fail` will work any more.
There should be a pytest equivalent - maybe even just `pytest.fail`

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py [hunk 3]
```diff
@@ -233,53 +119,6 @@ def _create_storage_service(self, service_class, settings):
...
-                settings.REMOTE_STORAGE_ACCOUNT_NAME,
-                settings.REMOTE_STORAGE_ACCOUNT_KEY,
-                protocol=settings.PROTOCOL,
-            )
-        else:
-            print("REMOTE_STORAGE_ACCOUNT_NAME and REMOTE_STORAGE_ACCOUNT_KEY not set in test settings file.")
-        self._set_test_proxy(service, settings)
-        return service
-
     def assertNamedItemInContainer(self, container, item_name, msg=None):
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/queuetestcase.py]: Now that we have the chance - this function name could be refactored to the python snake casing

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py [hunk 4]
```diff
@@ -69,38 +44,15 @@ def __init__(self):
...
-        self.logger = logging.getLogger('azure.storage')
-        # enable logging if desired
-        self.configure_logging()
-
-    def configure_logging(self):
-        self.enable_logging() if self.settings.ENABLE_LOGGING else self.disable_logging()
+    def _get_oauth_queue_url(self, name):
+        return "https://{}.queue.core.windows.net".format(name)
+    
+    def _account_url (self, name):
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/queuetestcase.py]: Looks like this and the function above can be refactored into a simple function.

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py [hunk 5]
```diff
@@ -171,54 +105,6 @@ def _set_test_proxy(service, settings):
...
-            self.settings.PREMIUM_STORAGE_ACCOUNT_NAME
-        )
-
-    def _get_remote_account_url(self):
-        return "{}://{}.blob.core.windows.net".format(
-            self.settings.PROTOCOL,
-            self.settings.REMOTE_STORAGE_ACCOUNT_NAME
-        )
-
     def _create_storage_service(self, service_class, settings):
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/queuetestcase.py]: Is this function still in use?

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 6]
```diff
@@ -32,25 +35,22 @@
...
 class StorageQueueTest(QueueTestCase):
     def setUp(self):
         super(StorageQueueTest, self).setUp()
-
-        queue_url = self._get_queue_url()
-        credentials = self._get_shared_key_credential()
-        self.qsc = QueueServiceClient(account_url=queue_url, credential=credentials)
         self.test_queues = []
 
     def tearDown(self):
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue.py]: Now that we're no longer using unittest.Testcase - I don't think the self.setUp and self.tearDown are being used any more - all this code should now be incorporated in the preparers, or cleaned up in some other way.
If we are creating extra queues that need to be deleted - we can probably call this function explicitly in the test where needed.
```python
try:
    # test code
finally:
    self._clean_up()
```

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 7]
```diff
@@ -120,49 +131,59 @@ def test_create_queue_with_options(self):
...
-        queues = list(self.qsc.list_queues())
+        qsc = QueueServiceClient(self._account_url(storage_account.name), storage_account_key)
+        queue_client = self._get_queue_reference(qsc)
+        queue_client.create_queue()
+        queues = list(qsc.list_queues())
 
         # Asserts
         self.assertIsNotNone(queues)
-        self.assertTrue(len(self.test_queues) <= len(queues))
+        self.assertEqual(len(queues), 1)
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue.py]: Now that we're no longer unsing unittest, we need to move away from the `self.assert` syntax, e.g.:
assertTrue, assertEqual, assertIsNone, assertIsNotNone, assertFalse etc etc

In most cases we should be able to use just a basic assert statement:
```
assert len(queues) == 1
```

File: sdk/storage/azure-storage-queue/tests/test_queue.py (hunk 6 above)
Comment (annatisch) [reply sdk/storage/azure-storage-queue/tests/test_queue.py]: I just occurred to me that we're cleaning up the resource group afterwards - so left over queues shouldn't be a problem :)

File: sdk/storage/azure-storage-queue/tests/test_queue_async.py [hunk 8]
```diff
@@ -220,22 +201,20 @@ def test_list_queues(self):
         self.assertNotEqual('', queues1[0].name)
         # Asserts
         self.assertIsNotNone(queues2)
-        self.assertTrue(len(self.test_queues) - 3 <= len(queues2))
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue_async.py]: We should keep some kind of length assertion in the list tests

File: sdk/storage/azure-storage-queue/tests/test_queue.py (hunk 6 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue.py]: we are not using them anymore. i've removed setup and teardown

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py (hunk 5 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/queuetestcase.py]: removed it

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py (hunk 4 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/queuetestcase.py]: 🥇 for my stupidity

File: sdk/storage/azure-storage-queue/tests/queuetestcase.py (hunk 2 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/queuetestcase.py]: yup..it's pytest.fail :)

File: sdk/storage/azure-storage-queue/tests/test_queue_async.py (hunk 8 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue_async.py]: added one.

File: sdk/storage/azure-storage-queue/tests/test_queue.py (hunk 7 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue.py]: I'll script this across blob, queues and files onve i have green CIs for the other two. manually refactoring every assert statement doesn't feel very practical.

File: sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py [hunk 9]
```diff
@@ -39,27 +41,29 @@ def _assert_stats_unavailable(self, stats):
...
-        url = self._get_queue_url()
-        credential = self._get_shared_key_credential()
-        qsc = QueueServiceClient(url, credential=credential)
+        qsc = QueueServiceClient(self._account_url(storage_account.name), storage_account_key)
 
         # Act
-        stats = qsc.get_service_stats()
-
+        stats = qsc.get_service_stats(raw_response_hook=self.override_response_body_with_live_status)
+        print(stats.geo_replication)
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py]: Print statement

File: sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py [hunk 10]
```diff
@@ -39,27 +41,29 @@ def _assert_stats_unavailable(self, stats):
     def override_response_body_with_unavailable_status(response):
         response.http_response.text = lambda: SERVICE_UNAVAILABLE_RESP_BODY
 
+    @staticmethod
+    def override_response_body_with_live_status(response):
+        response.http_response.text = lambda: SERVICE_LIVE_RESP_BODY
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py]: Why do we have to overwrite the live status response? Does this not work live?

File: sdk/storage/azure-storage-queue/tests/test_queue_samples_authentication.py [hunk 11]
```diff
@@ -75,22 +71,24 @@ def test_auth_active_directory(self):
...
-    def test_auth_shared_access_signature(self):
+    @ResourceGroupPreparer()          
+    @StorageAccountPreparer(name_prefix='pyacrstorage')
+    def test_auth_shared_access_signature(self, resource_group, location, storage_account, storage_account_key):
         # SAS URL is calculated from storage key, so this test runs live only
-        if TestMode.need_recording_file(self.test_mode):
+        if not self.is_live:
             return
-
+        cs = self.connection_string(storage_account, storage_account_key)
```
Comment (annatisch) [inline sdk/storage/azure-storage-queue/tests/test_queue_samples_authentication.py]: These are our samples - so a more descriptive name like "connection_string" might be user-friendlier

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 12]
```diff
@@ -32,84 +35,78 @@
...
-    record,
     LogCaptured,
 )
 
 # ------------------------------------------------------------------------------
 TEST_QUEUE_PREFIX = 'pythonqueue'
 
+FAKE_STORAGE = FakeStorageAccount(
+    name='pyacrstorage',
+    id='')
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue.py]: we can actually remove this

File: sdk/storage/azure-storage-queue/tests/test_queue.py [hunk 13]
```diff
@@ -176,17 +183,19 @@ def test_list_queues_with_options(self):
         assert generator1.location_mode is not None
         # Asserts
         self.assertIsNotNone(queues2)
-        self.assertTrue(len(self.test_queues) - 3 <= len(queues2))
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue.py]: added again.

File: sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py (hunk 10 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue_service_stats.py]: Creating a secondary location and making it available is either taking a lot of time or doesn't work live.

File: sdk/storage/azure-storage-queue/tests/test_queue_samples_authentication.py (hunk 11 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-queue/tests/test_queue_samples_authentication.py]: updated...

**Pull request merged**